### ISSUE TITLE
ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 0

### ISSUE THREAD

**Issue by fprochazka** [skjolber/3d-bin-container-packing#689]
Hi, for the following inputs

```java
	@Test
	public void test() throws Exception {
		DefaultContainer container = Container.newBuilder()
			.withDescription("1")
			.withSize(18, 12, 12)
			.withMaxLoadWeight(100000)
			.withEmptyWeight(0)
			.build();

		StackableItem b1 = new StackableItem(
			Box.newBuilder()
				.withId("b1")
				.withDescription("b1")
				.withSize(22, 5, 15)
				.withWeight(5)
				.withRotate3D()
				.build(),
			1
		);

		Packager packager = BruteForcePackager.newBuilder()
			.build();

		packager.newResultBuilder()
			.withContainers(ContainerItem.newListBuilder()
				.withContainer(container)
				.build())
			.withStackables(b1)
			.withDeadline(60_000)
			.build()
			.getContainers();
	}
```

the packager throws unexpected exception:

<details>
  <summary><code>java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 0</code></summary>

```
	at com.github.skjolber.packing.iterator.DefaultPermutationRotationIterator.get(DefaultPermutationRotationIterator.java:245)
	at com.github.skjolber.packing.iterator.DefaultPermutationRotationIterator.calculateMinStackableVolume(DefaultPermutationRotationIterator.java:83)
	at com.github.skjolber.packing.iterator.DefaultPermutationRotationIterator.<init>(DefaultPermutationRotationIterator.java:65)
	at com.github.skjolber.packing.iterator.DefaultPermutationRotationIterator$Builder.build(DefaultPermutationRotationIterator.java:25)
	at com.github.skjolber.packing.packer.bruteforce.BruteForcePackager$BruteForceAdapter.<init>(BruteForcePackager.java:87)
	at com.github.skjolber.packing.packer.bruteforce.BruteForcePackager.adapter(BruteForcePackager.java:161)
	at com.github.skjolber.packing.packer.AbstractPackager.pack(AbstractPackager.java:147)
	at com.github.skjolber.packing.packer.bruteforce.BruteForcePackagerResultBuilder.build(BruteForcePackagerResultBuilder.java:49)
```
</details>

This obviously doesn't have solution, but IMHO it should not throw exceptions, but instead cleanly return.

One of the problems is in `com.github.skjolber.packing.iterator.AbstractPermutationRotationIteratorBuilder#toMatrix`, which returns array with null items, which then causes problems in `com.github.skjolber.packing.iterator.DefaultPermutationRotationIterator#get`. I wasn't able to investigate further so far

**Comment by tyrcho** [comment]
I noticed the same bug today, here is a simple PR which reproduces the issue (similar to what @fprochazka wrote)
https://github.com/skjolber/3d-bin-container-packing/pull/691/files

**Comment by skjolber** [comment]
Fixed for the single-threaded brute force packagers, but not for the parallel packager yet (out of time for today). 

I've released this from the [3.x branch](https://github.com/skjolber/3d-bin-container-packing/tree/3.x) - try version 3.0.4.

**Comment by fprochazka** [comment]
@skjolber thank you! I confirm that `3.0.4` has fixed my problem

**Comment by skjolber** [comment]
@tyrcho and @fprochazka more important fixes in 3.0.5

**Comment by skjolber** [comment]
@tyrcho and @fprochazka finally 3.0.6 improves multi-container results and also halts impossible packaging jobs earlier.
